pkg/util/helper/index.go

```diff
@@ -18,15 +18,53 @@ package helper
 
 import (
 	"context"
+	"sync"
 
 	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
 	workv1alpha2 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha2"
+	"github.com/karmada-io/karmada/pkg/util"
 )
 
+const (
+	// IndexNameForResourceBindingClusters used for building indexer for resourcebinding object based on the field .spec.clusters
+	IndexNameForResourceBindingClusters = "resourcebinding.spec.clusters"
+	// IndexNameForClusterResourceBindingClusters used for building indexer for clusterresourcebinding object based on the field .spec.clusters
+	IndexNameForClusterResourceBindingClusters = "clusterresourcebinding.spec.clusters"
+)
+
+var (
+	registerOnce sync.Once
+)
+
+// Options Optional when defining the registration indexer
+type Options struct {
+	// EnableTaintManager If set to true will register resourcebinding.spec.clusters and clusterresourcebinding.spec.clusters indexer
+	EnableTaintManager bool
+}
+
+// RegisterFieldIndexes register all field indexes
+func RegisterFieldIndexes(ctx context.Context, mgr ctrl.Manager, opt *Options) error {
```

Comment:
I don't think it's a good idea to couple controller options.
I think each index function should own one sync.Once, so that it can be involved everywhere need it.

Reply:
This is an option. I refer to openkruise, which chooses to register all indexers into one function.

https://github.com/openkruise/kruise/blob/14d9ebdbfc248a86b459702fae1a228987a6674a/pkg/util/fieldindex/register.go#L43-L99

Reply:
Thanks for the information. 
What if the other component other than `karmada-controller-manager` also needs to register indexers? Note that not all components share the same set of indexers.

Reply:
> Note that not all components share the same set of indexers.

May be a  good suggestion is to create an index registration function for each component that needs to create indexers.